blog/_posts/2021-02-16-improving-scala-library-ecosystem.md

```diff
@@ -0,0 +1,127 @@
+
+Unfortunately, it is not an easy task to know whether a change on public API broke source or binary compatibility. Scala Center contracted Alexandre Archambault to create [sbt-version-policy][5]. This plugin helps library authors to self-check a version scheme. To use this, add the following to your `project/plugins.sbt`:
+
+~~~ scala
+addSbtPlugin("ch.epfl.scala" % "sbt-version-policy" % "1.0.0-RC5")
+~~~
+
+And declare your compatibility intention regarding the next release, in your `build.sbt`:
+
+~~~
+// The next release must be binary and source compatible with the previous release (it will be a patch release)
+ThisBuild / versionPolicyIntention := Compatibility.BinaryAndSourceCompatible
+~~~
+
+This plugin provides a task called `versionPolicyCheck`, which you can call in the CI (continuous integration) server. The task performs an automatic binary compatibility check.
+
+~~~ text
+> versionPolicyCheck
+~~~
+
+I think Early Semver gives better flexibility to both library authors and library users since it gives more information about what would be in minor upgrades (I often call this feature release) vs patch. For example, sbt 1 ships bug fixes as 1.3.x patch releases without going through RC cycle, so bug fixes are released quickly.
+
+During the minor upgrade (feature release), we aggressively add new features, while maintaining binary compatibility for the plugin ecosystem. These only come out once a year, and it goes through the RC cycle.
+
+## versionScheme, libraryDependencySchemes, and sbt 1.5.0
+
+As mentioned earlier, sbt contains a built-in eviction warning feature, but there are too many false positives right now because it needs to guess.
+
+Since sbt 1.4.0, there is a new setting `versionScheme`, which can be used by library authors as follows in `build.sbt`:
+
+~~~ scala
+ThisBuild / versionScheme := Some("early-semver")
+~~~
+
+sbt 1.4.0 includes this information into `pom.xml` and `ivy.xml` as a property. In addition, sbt uses the information to take the guessing out of eviction warning when this information is available.
+
+In sbt 1.5.0, eviction warnings will be replaced with [eviction errors][8]. Since we can now reliably detect whether two dependencies with different versions are compatible, or if they conflict, the build will fail if an incompatibility is detected in your dependencies.
+
+It might take a few years for the `versionScheme` information to become prevalent in the ecosystem, but once it happens the eviction warning could become more accurate. In the meantime, as a user of libraries you can manually configure the versioning scheme used by your libraries by using a new setting, `libraryDependencySchemes`. For instance, here is how we can tell sbt that Circe’s artifacts follow the Early SemVer scheme:
+
+~~~ scala
+ThisBuild / libraryDependencySchemes += "io.circe" %% "circe-*" % "early-semver"
+~~~
```

Comment:
And then ... what happens? The post gives me things I can configure, but it doesn't tell/show me what the result of doing so is. Consider adding a few more examples of actually firing up the commands that will perform the checks in a few situations, both when it errors out and when it succeeds.